goshopify.go

```diff
@@ -343,8 +343,19 @@ func (c *Client) doGetHeaders(req *http.Request, v interface{}) (http.Header, er
 	c.attempts = 0
 	c.logRequest(req)
 
+	// copy request body so it can be re-used
+	var body []byte
+	if req.Body != nil {
+		body, err = ioutil.ReadAll(req.Body)
+		defer req.Body.Close()
+	}
+	if err != nil {
```

Comment:
this should probably go up into the if block?

Reply:
Yes, I'll move it